File: string-handling/string-position/show_position.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine. 
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file. 
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Compile the program
4. Run the program
*/

using System;


namespace StringHandling
{

    class StringPosition
    {

        // The Main method is the entry point for all C# programs
        public static void Main(string[] args) {
            ShowPosition();
        }


        // Demonstrates how to find a character within a string
        public static void ShowPosition() {

            string subject = "Computer Science";
            int position = subject.IndexOf("x");
            Console.WriteLine(position);

        }
        

    }
}

File: string-handling/string-position/python/show_position.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

def main():
    """Demonstrates finding a letter within a string"""
    subject = "Computer Science"
    letter_sought = "m"
    show_position(subject, letter_sought)
    position = subject.find(letter_sought)
    print(f"{letter_sought} found at: {position}")
    

# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
